services/movie_session.py

```diff
@@ -0,0 +1,51 @@
+import datetime
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime.datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    movie = Movie.objects.get(id=movie_id)
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie,
+        cinema_hall=cinema_hall
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet | MovieSession:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_id: int) -> Optional[MovieSession]:
+    return MovieSession.objects.get(id=movie_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime.datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> Optional[MovieSession]:
+    session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(id=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    session.save()
+    return session
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    MovieSession.objects.get(id=session_id).delete()
```

Comment:
The `delete_movie_session_by_id` function uses `MovieSession.objects.get(id=session_id)`, which will raise a `DoesNotExist` exception if the session ID is not found. Consider handling this exception to avoid runtime errors.